src/MediatR/Mediator.cs

```diff
@@ -37,48 +39,76 @@ public TResponse Send<TResponse>(IRequest<TResponse> request)
         {
             var defaultHandler = GetHandler(request);
 
-            var result = defaultHandler.Handle(request);
+            var pipeline = GetPipeline(request, () => Task.FromResult(defaultHandler.Handle(request)));
 
-            return result;
+            return pipeline.Result;

```

Comment:
Don't block this way, sooner or later will cause a deadlock. I'd use `Task.Run(() => ...).Result` instead